### ISSUE TITLE
Extra layer encoder_output_to_decoder_dim cause issue with distributed training

### ISSUE THREAD

**Issue by ncoop57** [lucidrains/RETRO-pytorch#20]
Hiya, hope Ice Cream is doing well, as well as you of course!

I've been trying to get distributed training working with your library and I discovered this additional Linear layer `encoder_output_to_decoder_dim` not being used any where:

https://github.com/lucidrains/RETRO-pytorch/blob/main/retro_pytorch/retro_pytorch.py#L491

It seems to be a copy of the layer defined right above it `to_decoder_model_dim`, which does get used. Having this extra layer that is not part of the loss calculation causes the following error with data parallelism:

```
[RuntimeError: Expected to have finished reduction in the prior iteration before starting a new one.](https://github.com/pytorch/pytorch/issues/43259#)
````

Not sure if this layer is supposed to be there and it just didn't get used or if it is there by accident, so wanted to ask 🤓

**Comment by lucidrains** [comment]
<img width="575" alt="Screen Shot 2022-05-04 at 3 45 31 PM" src="https://user-images.githubusercontent.com/108653/166837533-058a3a1d-95d6-44d2-9765-d1c2b0f118cc.png">

thanks for asking about Ice Cream 😄 she is well!

and yes, that is indeed redundant because the code was changed for the encoder to directly output to the decoder dimensions https://github.com/lucidrains/RETRO-pytorch/blob/main/retro_pytorch/retro_pytorch.py#L518 

let me go get that fixed! 🙏

**Comment by lucidrains** [comment]
https://github.com/lucidrains/RETRO-pytorch/commit/f1662628d2d6a1e80919042916837d046ca882d6
